### PR TITLE
view all categories

### PR DESCRIPTION
**Pull request by DougSilvera**
#### Changes Made
1. Added file `CategoryController.cs` to `Controller` directory.
2. Added file `Category.cs` to `Models`
3. Added files `CategoryRepository.cs` and `ICategoryRepository` to `repositories`
4. Added file `categoryList.js` to `src/components/Categories`
5. Added file `categoryManager.js` to `src/modules`
1. Modified file `Header.js` to include navigation links to category page for logged in users.
2. Modified file `ApplicationViews.js` to include routes for category list module
​
#### Steps to Review
1. Checkout this branch locally.
    ```
    git fetch --all
    git checkout 39_view_all_categories
    ```
6. Open a new Terminal tab (⌘T) and navigate to the server directory.
7. Test app functionality.
    > When a logged in user clicks on the Category Management link they are taken to a list of categories and each entry in the list has an "edit" and "delete" link option.
8. View code file.
    > Confirm file modifications are present as indicated above.
    > Confirm no unused code or extraneous comments exist.

### CODE REVIEW

**Review by JacobF94 — COMMENTED** [review]
Tested functionality, all successful. No conflicts, all tags were on the page

File: Tabloid/Controllers/CategoryController.cs [hunk 1]
```diff
@@ -0,0 +1,54 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using System;
+using Tabloid.Models;
+using Tabloid.Repositories;
+
+
+namespace Tabloid.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
```
Comment (joshdbarton) [inline Tabloid/Controllers/CategoryController.cs:13]: Should this controller be auth protected? should there be auth checks in any of the methods?

**Review by JacobF94 — APPROVED** [review]
re-doing my review so it is an approval

File: Tabloid/client/src/components/ApplicationViews.js [hunk 2]
```diff
@@ -12,6 +13,9 @@ export default function ApplicationViews({ isLoggedIn }) {
         <Route path="/" exact>
           {isLoggedIn ? <Hello /> : <Redirect to="/login" />}
         </Route>
+        <Route path="/categories">
```
Comment (joshdbarton) [inline Tabloid/client/src/components/ApplicationViews.js]: should this route be an `exact` match?

File: Tabloid/client/src/modules/categoryManager.js [hunk 3]
```diff
@@ -0,0 +1,7 @@
+const _apiUrl = "/api/category";
+
+
+export const getAllCategories = () => {
+    return fetch(_apiUrl)
```
Comment (joshdbarton) [inline Tabloid/client/src/modules/categoryManager.js]: If you do require auth to do this action, you'll need to get a token first and pass it into the headers of this request.

**Review by joshdbarton — CHANGES_REQUESTED** [review]
reviewed, take a look at my comments

File: Tabloid/client/src/components/ApplicationViews.js (hunk 2 above)
Comment (DougSilvera) [reply Tabloid/client/src/components/ApplicationViews.js]: added exact route path in commit "resolved exact route path"

File: Tabloid/client/src/modules/categoryManager.js (hunk 3 above)
Comment (DougSilvera) [reply Tabloid/client/src/modules/categoryManager.js]: refactored function to include passing token to api for authorization

File: Tabloid/Controllers/CategoryController.cs (hunk 1 above)
Comment (DougSilvera) [reply Tabloid/Controllers/CategoryController.cs:13]: auth protected controller, will add auth checks to methods to verify user type permissions when that feature is added

**Review by joshdbarton — APPROVED** [review]

**Pull request merged**